src/services/commandExecutor.ts

```diff
@@ -43,11 +43,17 @@ export interface IGithubCommandExecutor {
 }
 
 export class ShellCommandExecutor implements ICommandExecutor {
-  async execute(commands: string[]): Promise<CommandExecutorResponse> {
+  async execute(commands: string[], cwd?: string): Promise<CommandExecutorResponse> {
     const exec = promisify(cp.exec);
     const resp = new CommandExecutorResponse();
     try {
-      const { stdout, stderr } = await exec(commands.join(' && '), { maxBuffer: c.get('MAX_STDOUT_BUFFER_SIZE') });
+      const options: { maxBuffer: number; cwd?: string } = {
+        maxBuffer: c.get('MAX_STDOUT_BUFFER_SIZE'),
```

Comment:
dumb nitpick: if you wanted, you could add cwd within the object declaration.
`...(cwd ? { cwd }: {}),` or `...(!!cwd && { cwd })`
But up to you, because your code might be more clear in the long run.

Reply:
After looking at the node docs for [exe](https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback) I thnk it might be a cleaner and safer approach to declare it outside of the object, because when no `cwd` passed in as a param then the option will just be the `maxBuffer`, leaving the `cwd` in a default state which is the current working directory `.` The `cwd` option can either be a string or a URL. Let me know if that makes sense.

Reply:
I think either way works!! No worries on this at all!